### ISSUE TITLE
test failures on macos

### ISSUE THREAD

**Issue by michaelkirk** [georust/gdal#83]
Hi, I'm new to this repository. 👋 

I just built from master and am seeing these test failures:
```
running 70 tests
test config::tests::test_set_option_with_embedded_nul ... ok
test config::tests::test_clear_option ... ok
test config::tests::test_set_get_option ... ok
test raster::tests::test_get_description ... ok
test raster::tests::test_get_driver_by_name ... ok
test raster::tests::test_create_with_band_type ... ok
test raster::tests::test_create ... ok
test raster::tests::test_get_band_type ... ok
test raster::tests::test_geo_transform ... ok
test raster::tests::test_get_metadata_item ... ok
test raster::tests::test_get_raster_block_size ... ok
test raster::tests::test_create_copy ... ok
test raster::tests::test_get_no_data_value ... ok
test raster::tests::test_get_raster_count ... ok
test raster::tests::test_get_rasterband ... ok
test raster::tests::test_get_projection ... ok
test raster::tests::test_get_offset ... FAILED
test raster::tests::test_get_dataset_driver ... ok
test raster::tests::test_get_rasterband_block_size ... ok
test raster::tests::test_get_rasterband_size ... ok
test raster::tests::test_get_raster_size ... ok
test raster::tests::test_get_scale ... FAILED
test raster::tests::test_set_no_data_value ... ok
test raster::tests::test_set_metadata_item ... ok
test raster::tests::test_write_raster ... ok
test raster::tests::test_open ... ok
test raster::tests::test_read_full_raster_as ... ok
test raster::tests::test_read_raster ... ok
test raster::tests::test_read_raster_as ... ok
test spatial_ref::tests::authority ... ok
test spatial_ref::tests::from_esri_to_proj4 ... ok
test spatial_ref::tests::from_epsg_to_wkt_proj4 ... FAILED
test spatial_ref::tests::auto_identify ... ok
test vector::geometry::tests::test_is_empty ... ok
test vector::geometry::tests::test_area ... ok
test vector::ops::geometry::intersection::tests::test_intersection_no_gdal_ptr ... ok
test spatial_ref::tests::from_wkt_to_proj4 ... ok
test vector::ops::geometry::intersection::tests::test_intersection_no_intersects ... ok
test vector::tests::convert_geo::test_import_export_geometrycollection ... ok
test spatial_ref::tests::comparison ... ok
test vector::ops::geometry::intersection::tests::test_intersection_success ... ok
test vector::tests::convert_geo::test_import_export_linestring ... ok
test vector::tests::convert_geo::test_import_export_multipoint ... ok
test vector::tests::convert_geo::test_import_export_multilinestring ... ok
test vector::tests::convert_geo::test_import_export_multipolygon ... ok
test vector::tests::convert_geo::test_import_export_polygon ... ok
test vector::tests::convert_geo::test_import_export_point ... ok
test vector::tests::test_convex_hull ... ok
test vector::tests::test_create_bbox ... ok
test spatial_ref::tests::from_proj4_to_wkt ... FAILED
test vector::geometry::tests::test_spatial_reference ... ok
test vector::tests::test_float_field ... ok
test vector::tests::test_get_layer_by_name ... ok
test vector::tests::test_json ... ok
test vector::tests::test_geom_fields ... FAILED
test vector::tests::test_geom_accessors ... ok
test vector::tests::test_iterate_features ... ok
test spatial_ref::tests::transform_ogr_geometry ... FAILED
test vector::tests::test_layer_count ... ok
test vector::tests::test_layer_spatial_reference ... ok
test vector::tests::test_layer_extent ... ok
test vector::tests::test_schema ... ok
test version::tests::test_version_info ... ok
test vector::tests::test_string_field ... ok
test vector::tests::test_missing_field ... ok
test spatial_ref::tests::transform_coordinates ... FAILED
test vector::tests::test_spatial_filter ... ok
test spatial_ref::tests::failing_transformation ... ok
test vector::tests::test_wkt ... ok
test vector::tests::test_write_features ... ok

failures:

---- raster::tests::test_get_offset stdout ----
thread 'raster::tests::test_get_offset' panicked at 'assertion failed: `(left == right)`
  left: `None`,
 right: `Some(0.0)`', src/raster/tests.rs:347:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- raster::tests::test_get_scale stdout ----
thread 'raster::tests::test_get_scale' panicked at 'assertion failed: `(left == right)`
  left: `None`,
 right: `Some(1.0)`', src/raster/tests.rs:339:5

---- spatial_ref::tests::from_epsg_to_wkt_proj4 stdout ----
thread 'spatial_ref::tests::from_epsg_to_wkt_proj4' panicked at 'assertion failed: `(left == right)`
  left: `"GEOGCS[\"WGS 84\",DATUM[\"WGS_1984\",SPHEROID[\"WGS 84\",6378137,298.257223563,AUTHORITY[\"EPSG\",\"7030\"]],AUTHORITY[\"EPSG\",\"6326\"]],PRIMEM[\"Greenwich\",0,AUTHORITY[\"EPSG\",\"8901\"]],UNIT[\"degree\",0.0174532925199433,AUTHORITY[\"EPSG\",\"9122\"]],AUTHORITY[\"EPSG\",\"4326\"]]"`,
 right: `"GEOGCS[\"WGS 84\",DATUM[\"WGS_1984\",SPHEROID[\"WGS 84\",6378137,298.257223563,AUTHORITY[\"EPSG\",\"7030\"]],AUTHORITY[\"EPSG\",\"6326\"]],PRIMEM[\"Greenwich\",0,AUTHORITY[\"EPSG\",\"8901\"]],UNIT[\"degree\",0.0174532925199433,AUTHORITY[\"EPSG\",\"9122\"]],AXIS[\"Latitude\",NORTH],AXIS[\"Longitude\",EAST],AUTHORITY[\"EPSG\",\"4326\"]]"`', src/spatial_ref/tests.rs:39:5

---- spatial_ref::tests::from_proj4_to_wkt stdout ----
thread 'spatial_ref::tests::from_proj4_to_wkt' panicked at 'assertion failed: `(left == right)`
  left: `"PROJCS[\"unknown\",GEOGCS[\"unknown\",DATUM[\"Unknown based on GRS80 ellipsoid\",SPHEROID[\"GRS 1980\",6378137,298.257222101,AUTHORITY[\"EPSG\",\"7019\"]]],PRIMEM[\"Greenwich\",0,AUTHORITY[\"EPSG\",\"8901\"]],UNIT[\"degree\",0.0174532925199433,AUTHORITY[\"EPSG\",\"9122\"]]],PROJECTION[\"Lambert_Azimuthal_Equal_Area\"],PARAMETER[\"latitude_of_center\",52],PARAMETER[\"longitude_of_center\",10],PARAMETER[\"false_easting\",4321000],PARAMETER[\"false_northing\",3210000],UNIT[\"metre\",1,AUTHORITY[\"EPSG\",\"9001\"]],AXIS[\"Easting\",EAST],AXIS[\"Northing\",NORTH]]"`,
 right: `"PROJCS[\"unnamed\",GEOGCS[\"GRS 1980(IUGG, 1980)\",DATUM[\"unknown\",SPHEROID[\"GRS80\",6378137,298.257222101]],PRIMEM[\"Greenwich\",0],UNIT[\"degree\",0.0174532925199433]],PROJECTION[\"Lambert_Azimuthal_Equal_Area\"],PARAMETER[\"latitude_of_center\",52],PARAMETER[\"longitude_of_center\",10],PARAMETER[\"false_easting\",4321000],PARAMETER[\"false_northing\",3210000],UNIT[\"Meter\",1]]"`', src/spatial_ref/tests.rs:28:5

---- vector::tests::test_geom_fields stdout ----
thread 'vector::tests::test_geom_fields' panicked at 'assertion failed: geom_field.spatial_ref().unwrap() == spatial_ref2', src/vector/tests/mod.rs:216:5

---- spatial_ref::tests::transform_ogr_geometry stdout ----
thread 'spatial_ref::tests::transform_ogr_geometry' panicked at 'assertion failed: `(left == right)`
  left: `"POLYGON ((5509543.1508097 1716062.19161922,5467122.00033 1980151.20428024,5623571.02849272 2010213.31025368,5671834.92154436 1746968.07828026,5509543.1508097 1716062.19161922))"`,
 right: `"POLYGON ((7166004.81833779 564241.992615556,7403258.72613859 654346.74777158,7345135.05829164 847853.579444857,7112561.07890564 758804.633233282,7166004.81833779 564241.992615556))"`', src/spatial_ref/tests.rs:115:5

---- spatial_ref::tests::transform_coordinates stdout ----
thread 'spatial_ref::tests::transform_coordinates' panicked at 'assertion failed: f > 0.99999', src/lib.rs:39:5


failures:
    raster::tests::test_get_offset
    raster::tests::test_get_scale
    spatial_ref::tests::from_epsg_to_wkt_proj4
    spatial_ref::tests::from_proj4_to_wkt
    spatial_ref::tests::transform_coordinates
    spatial_ref::tests::transform_ogr_geometry
    vector::tests::test_geom_fields

test result: FAILED. 63 passed; 7 failed; 0 ignored; 0 measured; 0 filtered out
```

Travis looks to be passing all the tests, running gdal 2.0. 

I'm using gdal 3.1.2 (installed via [homebrew](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb)).

Is it expected for the tests to fail?

**Comment by rmanoka** [comment]
@michaelkirk I've tested this crate with gdal 2.4.2 and all the tests pass.   I believe the osgeo/gdal changes are triggering your fails.  For instance, I believe, the first two:  offset, scale were probably fixed "upstream" around this commit.

```
commit 69f25f253d141faf836c400676f9f94dd3f43707
Author: Even Rouault <even.rouault@spatialys.com>
Date:   Fri Jul 12 18:42:23 2019 +0200

    PAMRasterBand: add presence flag for Offset and SetScale so that GetOffset()/GetScale() properly reports if they are defined in PAM
```

**Comment by michaelkirk** [comment]
Indeed it seems to be a difference between gdal 2 vs. 3.

> Do you know how to setup CI for this repo, where we test on both gdal 2.x and gdal 3.x?

Having a go at this now. Hopefully you'll see a PR from me soon!

**Comment by jdroenner** [comment]
is this fixed with the CI changes (#86)?
